### PR TITLE
feat(earnings): integrate ui with API

### LINKED ISSUE #179: Implement earnings tabular list and filter
https://www.figma.com/file/G4Kl9fzLRgnk0SaeTne1Mc/Admin-Panel---Master?node-id=6255%3A3408

For a first task we want to implement the earnings/default and earnings/filters view.  Implement tabular data display, and filtering of tabular data.   Download/upload is not part of this issue.

Additional notes:
- implement sort by column (requires extension of API by @Kpoke )
- paging through data (requires query to return total count of records, @Kpoke )

### PR DESCRIPTION
**Pull request by Mloweedgar**
## Description

Integrates Earnings UI with the earnigs API

**Issue(s) addressed**

- Resolves #179 

**What kind of change(s) does this PR introduce?**

- [x] Enhancement
- [ ] Bug fix
- [ ] Refactor

**Please check if the PR fulfills these requirements**

- [x] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## Issue

**What is the current behavior?**
Earnings Table displays dummy data

**What is the new behavior?**
Earning table displays data returned from earnings API, and the following features are available

1. Paginate results displayed in table
2. Change number of rows displayed per page(20,50,100)
3. Sort table rows based on sortable columns(Grower, Funder, Amount)
4. Filter Earnings 
4. select and view details of earning
4. Export earnings to CSV

Displaying Payments in tabular list

1.  Paginate payments displayed in the table
2. Change number of rows displayed per page(20,50,100)
3. Filter payments 
3. select and view payment details
3. Sort table rows based on sortable columns(Grower, Funder, Amount)
4. Upload processed Payments in CSV format

## Breaking change

**Does this PR introduce a breaking change?**

NO breaking changes

### CODE REVIEW

File: src/components/Routers.js [hunk 1]
```diff
@@ -41,14 +41,32 @@ export default function Routers() {
...
+                      component={
+                        child?.disabled ? Unauthorized : child?.component
+                      }
+                      exact={child?.exact}
+                      getScrollContainerRef={
+                        child?.linkTo === '/verify'
+                          ? () => refContainer.current
+                          : undefined
+                      }
+                      key={`route_${i}`}
```
Comment (pieroapretto) [inline src/components/Routers.js:58]: Is using index to set `key` values good practice?

Comment (pieroapretto) [reply src/components/Routers.js:58]: Sure. However, using index values to generate unique keys could negatively impact performance and may cause issues with component state according to React [documentation](https://reactjs.org/docs/lists-and-keys.html).

**Review by Siddhant-K-code — APPROVED** [review]
LGTM 🎉 

Thanks ❤️

File: .env.development [hunk 2]
```diff
@@ -3,3 +3,6 @@ REACT_APP_API_ROOT=https://dev-k8s.treetracker.org/api/admin
 REACT_APP_TREETRACKER_API_ROOT=https://dev-k8s.treetracker.org/treetracker
 REACT_APP_ENABLE_CAPTURE_MATCHING=true
 REACT_APP_ENABLE_EARNINGS=true
+REACT_APP_ENABLE_PAYMENTS=true
+REACT_APP_API_URL=https://dev-k8s.treetracker.org
+REACT_APP_EARNINGS_API_MAPPING=/earnings
```
[the remark this answers is not in the document]
Comment (Siddhant-K-code) [reply .env.development:8]: @Mloweedgar Just add this line

**Review by Siddhant-K-code — CHANGES_REQUESTED** [review]
Fix Merge Conflicts & solve UnResolved

File: src/utilities/index.js [hunk 3]
```diff
@@ -22,3 +24,44 @@ export const stringToSearchRegExp = (value) => `/.*${value}.*/i`;
...
+ * @name covertDateStringToHumanReadableFormat
+ * @description receives a date iso string and converts it to a human readable format
+ *
+ * @param {string} dateString - date iso string to convert
+ * @param {string} format - format to convert to
+ * @link https://github.com/felixge/node-dateformat - library used to convert date(follow link to see format options)
+ *
+ * @returns {string} human readable date
+ */
+export const covertDateStringToHumanReadableFormat = (
```
Comment (nmcharlton) [inline src/utilities/index.js:39]: I'd suggest using or expanding the existing utility functions in `src/common/locale.js` to achieve this.
That way, the user will always see a format appropriate to their locale.

**Comment by nmcharlton** [comment]
I've added a few more comments, but I'm happy to merge the PR and raise separate issues to capture them, especially if they require significant effort.

**Comment by Mloweedgar** [comment]
Thanks @nmcharlton , I see they are minor, I am going to address them here. I real do appreciate your feedback, thank you very much.

**Comment by nmcharlton** [comment]
Can you rename the handler functions and class names to match the new RESET button labels?

**Comment by Mloweedgar** [comment]
@nmcharlton I have updated the handler functions by renaming the `Cancel` word to `Reset`

**Review by nmcharlton — APPROVED** [review]

**Pull request merged**